src/goto-instrument/contracts/contracts.cpp

```diff
@@ -678,6 +681,12 @@ void code_contractst::apply_loop_contract(
   local_may_aliast local_may_alias(goto_function);
   natural_loops_mutablet natural_loops(goto_function.body);
 
+  if(!natural_loops.loop_map.size())
+    return;
+
+  goto_function_inline(
+    goto_functions, function_name, ns, log.get_message_handler());
```

Comment:
As discussed offline: calls to `goto_function_inline` will require `goto_functions.update()` to be called afterwards.